extensions/microsoft-authentication/src/utils.ts

```diff
@@ -2,7 +2,31 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
+import { env, UIKind, Uri } from 'vscode';
 
 export function toBase64UrlEncoding(base64string: string) {
 	return base64string.replace(/=/g, '').replace(/\+/g, '-').replace(/\//g, '_'); // Need to use base64url encoding
 }
+
+export function isSupportedEnvironment(uri: Uri): boolean {
+	if (env.uiKind === UIKind.Desktop) {
+		return true;
+	}
+	// local development (localhost:* or 127.0.0.1:*)
+	if (/^https?$/i.test(uri.scheme) && /^(localhost|127\.0\.0\.1):\d+$/i.test(uri.authority)) {
```

Comment:
you may want to use `isLocalHost` for this: https://github.com/microsoft/vscode/blob/8fa5722aad72c6712c821bc1deeede56966b5d84/src/vs/platform/tunnel/common/tunnel.ts#L149

Reply:
switched to using that. (this is in extension land so I can't leverage that directly)